src/app/common/loading-modal/loading-dialog.service.ts

```diff
@@ -0,0 +1,22 @@
+import {Injectable} from '@angular/core';
+import {MatSnackBar, MatSnackBarRef} from '@angular/material/snack-bar';
+import {LoadingModalComponent} from './loading-modal.component';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoadingDialogService {
```

Comment:
I would rename it to `LoadingIndicatorService` or `LoadingSnackbarService` to match the reality.